### PR TITLE
Fix compiler warnings from Qt Creator

### PR DESCRIPTION
**Pull request by fdelapena**
Use breaks in cases and modern casting to make the compiler happy.

Tested with Qt Creator 4.7.1 (Qt 3.11.1) and Clang 7.0.0 / GCC 8.2.1.

### CODE REVIEW

**Comment by Ghabry** [comment]
imo the pro file is deprecated, everything should migrate to CMake.

**Comment by fmatthew5876** [comment]
Can you still use Qt Creator without the pro file? If so I'd be in favor of getting rid of it too.

**Comment by Ghabry** [comment]
QtCreator has CMake support by now. The Windows build still uses the qmake system, can die when vcpkg is rolled out.

**Comment by fdelapena** [comment]
This should be ready for testing.

**Comment by Ghabry** [comment]
wow what kind of pedantic compiler is this complaining about (int) instead of static_cast\<int\>?

**Comment by fdelapena** [comment]
Yes, clang defaults seem to be a _bit_ pedantic for old style casts.

File: src/dialogevent.h [hunk 1]
```diff
@@ -13,15 +13,15 @@ class DialogEvent : public QDialog
...
+    explicit DialogEvent(QWidget *parent = nullptr);
     ~DialogEvent();
 
     static int edit(QWidget *parent, RPG::Event *event);
 
     static bool equalEvents(const RPG::Event &e1,
                             const RPG::Event &e2);
 
-    RPG::Event event() const;
+    RPG::Event getEvent() const;
```
Comment (Ghabry) [inline src/dialogevent.h]: any reason this was renamed? At least just lowercase without get-prefix is the Qt-convention for the getter.
Do we use get-prefixed getters in the other parts of the editor?

**Comment by Ghabry** [comment]
the problem with changing the old-style casts: As soon as there is another PR they will be partially reintroduced because nobody of us cares enough to reject them during a review ^^'

File: src/dialogevent.h (hunk 1 above)
Comment (fdelapena) [reply src/dialogevent.h]: There's some conflict with some Qt member called ::event if kept :confused:

**Comment by fdelapena** [comment]
> the problem with changing the old-style casts: As soon as there is another PR they will be partially reintroduced because nobody of us cares enough to reject them during a review ^^'

They may be amended occasionally, probably not a major issue, or maybe setting clang in that pedantic level in a build job to keep them a bit more visible from logs. I expect to take care of them over time.

**Pull request merged**